### PR TITLE
Add cross OS support for MAKEDLLNAME

### PR DESCRIPTION
**Pull request by sdmaclea**
During cross OS DAC build, it is expected that MAKEDLLNAME returns the `TARGET` DLL name.

Fix this on `Windows` `HOST`.

I didn't make the equivalent change for Unix because I don't have a use case to test it.

### CODE REVIEW

**Comment by jkotas** [comment]
This is one of those cases that can be both either HOST or TARGET.

Where is this needed for the cross-targeting builds?

**Comment by sdmaclea** [comment]
The DAC uses it via  `MAIN_CLR_DLL_NAME_*` to find the base CoreCLR library.

I can split `MAKEDLLNAME` to `MAKE_TARGET_DLLNAME` and `MAKE_HOST_DLLNAME`

**Comment by jkotas** [comment]
MAKE_TARGET_DLLNAME sounds reasonable.

I do not think we need `MAKE_HOST_DLLNAME`. Host is implicit for OS APIs.

**Comment by jkotas** [comment]
BTW: As I was reviewing this, I found that `MSCOREE_SHIM_W/_A` is dead code (left overs from desktop). If you would like to delete it as part of this PR...

File: src/coreclr/src/utilcode/pedecoder.cpp [hunk 1]
```diff
@@ -3114,9 +3114,9 @@ BOOL PEDecoder::ForceRelocForDLL(LPCWSTR lpFileName)
     return TRUE;
 #else
 
-    // Contracts in ConfigDWORD do WszLoadLibrary(MSCOREE_SHIM_W).
+    // Contracts in ConfigDWORD do WszLoadLibrary(MAIN_CLR_DLL_NAME_W).
```
Comment (jkotas) [inline src/coreclr/src/utilcode/pedecoder.cpp]: This should be deleted instead.

File: src/coreclr/src/inc/palclr.h [hunk 2]
```diff
@@ -606,10 +606,27 @@
 #define MAKEDLLNAME_W(name) name W(".dll")
 #define MAKEDLLNAME_A(name) name  ".dll"
 
+// Target platform-specific library naming
+//
+#ifdef TARGET_WINDOWS
+#define MAKE_TARGET_DLLNAME_W(name) name W(".dll")
```
Comment (jkotas) [inline src/coreclr/src/inc/palclr.h]: Would it make sense to put this into `src\coreclr\src\inc>notepad crosscomp.h` where other similar definitions are? It should avoid the duplication between palclr.h and pal.h.

File: src/coreclr/src/utilcode/dacutil.cpp [hunk 3]
```diff
@@ -107,7 +107,7 @@ LiveProcDataTarget::GetImageBase(
     // about right now is the base of mscorwks.
     //
 
-    if (wcscmp(name, MAIN_CLR_DLL_NAME_W))
+    if (wcscmp(name, TARGET_MAIN_CLR_DLL_NAME_W))
```
Comment (jkotas) [inline src/coreclr/src/utilcode/dacutil.cpp]: Nit: This is the live instance - it does not need `TARGET_...`.

File: src/coreclr/src/inc/palclr.h (hunk 2 above)
Comment (sdmaclea) [reply src/coreclr/src/inc/palclr.h]: I was wondering where to put it.  Seems as good as any.

File: src/coreclr/src/utilcode/dacutil.cpp (hunk 3 above)
Comment (sdmaclea) [reply src/coreclr/src/utilcode/dacutil.cpp]: Ok I can revert it.

File: src/coreclr/src/pal/inc/pal.h [hunk 4]
```diff
@@ -4844,6 +4844,7 @@ class NativeExceptionHolderFactory
 #define MAKEDLLNAME(x) MAKEDLLNAME_A(x)
 #endif
 
+
```
Comment (jkotas) [inline src/coreclr/src/pal/inc/pal.h]: Nit: Unnecessary new line?

**Review by jkotas — APPROVED** [review]

**Pull request merged**